src/Cli/MergeConfig.cs

```diff
@@ -0,0 +1,126 @@
+                    MergeArrays(jsonWriter, root2);
+                }
+                else
+                {
+                    MergeObjects(jsonWriter, root1, root2);
+                }
+            }
+
+            return Encoding.UTF8.GetString(outputBuffer.WrittenSpan);
+        }
+
+        /// <summary>
+        /// This helper methods tries to merge two Json Object recursively.
+        /// </summary>
+        private static void MergeObjects(Utf8JsonWriter jsonWriter, JsonElement root1, JsonElement root2)
+        {
+            jsonWriter.WriteStartObject();
+
+            // If a property exists in both documents, either:
+            // * Merge them, if the value kinds match (e.g. both are objects or arrays) recursively,
+            // * Completely override the value of the first with the one from the second, if the value kind mismatches (e.g. one is object, while the other is an array or string),
+            // * Or favor the value of the first, if the second one is null.
+            foreach (JsonProperty property in root1.EnumerateObject())
+            {
+                string propertyName = property.Name;
+
+                JsonValueKind newValueKind;
+
+                if (root2.TryGetProperty(propertyName, out JsonElement newValue) && (newValueKind = newValue.ValueKind) != JsonValueKind.Null)
+                {
+                    jsonWriter.WritePropertyName(propertyName);
+
+                    JsonElement originalValue = property.Value;
+                    JsonValueKind originalValueKind = originalValue.ValueKind;
+
+                    if (newValueKind == JsonValueKind.Object && originalValueKind == JsonValueKind.Object)
+                    {
+                        MergeObjects(jsonWriter, originalValue, newValue); // Recursive call
+                    }
+                    else if (newValueKind == JsonValueKind.Array && originalValueKind == JsonValueKind.Array)
+                    {
+                        MergeArrays(jsonWriter, newValue);
+                    }
+                    else
+                    {
+                        newValue.WriteTo(jsonWriter);
+                    }
+                }
+                else
+                {
+                    property.WriteTo(jsonWriter);
+                }
+            }
+
+            // Write all the properties of the second document that are unique to it.
+            foreach (JsonProperty property in root2.EnumerateObject())
+            {
+                if (!root1.TryGetProperty(property.Name, out _))
+                {
+                    property.WriteTo(jsonWriter);
+                }
+            }
+
+            jsonWriter.WriteEndObject();
+        }
+
+        /// <summary>
+        /// This helper methods tries to merge two Json Array such that the values of second array
+        /// completely overrides that of the first.
+        /// </summary>
+        private static void MergeArrays(Utf8JsonWriter jsonWriter, JsonElement root2)
```

Comment:
Is it actually merging two arrays? Looks like it is just appending the array from the overriding json. A better function name maybe - `AppendArrayToJsonObject` ?

Reply:
No, we are not appending. We are completely overriding, the merged array won't have any elements from first array.

Reply:
I mean `MergeArrays` gives the feel that we are merging two arrays when we are not doing so. And looking at the function, we are just writing every `element` in the array to `jsonWriter`. Feels like appending.